dev/archery/archery/crossbow/reports.py

```diff
@@ -173,10 +173,15 @@ def url(self, query):
         repo_url = self.job.queue.remote_url.strip('.git')
         return '{}/branches/all?query={}'.format(repo_url, query)
 
+    def branch_url(self, branch):
+        repo_url = self.job.queue.remote_url.strip('.git')
```

Comment:
This isn't correct:
```python
>>> "something".strip(".git")
'somethin'
```

Starting with Python 3.9 you can use [`str.removesuffix`](https://docs.python.org/3/library/stdtypes.html#str.removesuffix), but for now you'll need something like:
```python
@property
def repo_url(self):
    url = self.job.queue.remote_url
    return url[:-4] if url.endswith('.git') else url
```

Reply:
Thanks for that, I did copy&paste from the existing url extraction, I have modified the original one (https://github.com/apache/arrow/blob/master/dev/archery/archery/crossbow/reports.py#L173) to solve that issue too!